src/lib.rs

```diff
@@ -357,3 +399,867 @@ const fn convert_case(keysym: Keysym) -> (Keysym, Keysym) {
 
     (Keysym(upper), Keysym(lower))
 }
+
+// We don't use the uint32_t types here, to save some space.
+struct CodePair {
+    keysym: u16,
+    ucs: u16,
+}
+
+#[rustfmt::skip]
+const KEYSYMTAB: [CodePair; 763] = [
```

Comment:
Is this table automatically generated?

Reply:
In libxkbcommon, no, it's manual over there(hasn't changed there for a long). It's a manual table, and generating it will require pulling the libxkbcommon sources, translating the `.c` file. I'd generally just subscribe for libxkbcommon releases, but I don't think this table will change over time, unless someone will develop a new native language, because adding emojis is future compat.